File: src/main/kotlin/moe/kurenai/tdlight/model/chat/ChatJoinRequest.java
package moe.kurenai.tdlight.model.chat;

import com.fasterxml.jackson.annotation.JsonProperty;
import dev.tobee.telegram.model.message.User;

import java.util.Optional;

public record ChatJoinRequest(
        @JsonProperty("chat") Chat chat,
        @JsonProperty("from") User from,
        @JsonProperty("date") long date,
        @JsonProperty("bio") Optional<String> bio,
        @JsonProperty("invite_link") Optional<dev.tobee.telegram.model.chat.ChatInviteLink> inviteLink
) {
}

File: src/main/kotlin/moe/kurenai/tdlight/model/chat/ChatJoinRequest.kt
package moe.kurenai.tdlight.model.chat

import com.fasterxml.jackson.annotation.JsonProperty
import moe.kurenai.tdlight.model.message.User

data class ChatJoinRequest(

    @JsonProperty("chat")
    val chat: Chat,

    @JsonProperty("from")
    val from: User,

    @JsonProperty("date")
    val date: Long = 0,

    @JsonProperty("bio")
    val bio: String? = null,

    @JsonProperty("invite_link")
    val inviteLink: ChatInviteLink? = null,
) {
}